src/abstract_domains.rs

```diff
@@ -1644,6 +1654,125 @@ impl AbstractDomain for ExpressionDomain {
+                .refine_parameters(arguments)
+                .lt(&right.refine_parameters(arguments)),
+            ExpressionDomain::Mul { left, right } => left
+                .refine_parameters(arguments)
+                .mul(&right.refine_parameters(arguments)),
+            ExpressionDomain::MulOverflows {
+                left,
+                right,
+                result_type,
+            } => left
+                .refine_parameters(arguments)
+                .mul_overflows(&right.refine_parameters(arguments), result_type.clone()),
+            ExpressionDomain::Ne { left, right } => left
+                .refine_parameters(arguments)
+                .not_equals(&right.refine_parameters(arguments)),
+            ExpressionDomain::Neg { operand } => operand.refine_parameters(arguments).neg(),
+            ExpressionDomain::Not { operand } => operand.refine_parameters(arguments).not(),
+            ExpressionDomain::Offset { left, right } => left
+                .refine_parameters(arguments)
+                .offset(&right.refine_parameters(arguments)),
+            ExpressionDomain::Or { left, right } => left
+                .refine_parameters(arguments)
+                .or(&right.refine_parameters(arguments)),
+            ExpressionDomain::Reference(..) => self.clone(),
+            ExpressionDomain::Rem { left, right } => left
+                .refine_parameters(arguments)
+                .rem(&right.refine_parameters(arguments)),
+            ExpressionDomain::Shl { left, right } => left
+                .refine_parameters(arguments)
+                .shl(&right.refine_parameters(arguments)),
+            ExpressionDomain::ShlOverflows {
+                left,
+                right,
+                result_type,
+            } => left
+                .refine_parameters(arguments)
+                .shl_overflows(&right.refine_parameters(arguments), result_type.clone()),
+            ExpressionDomain::Shr { left, right } => left
+                .refine_parameters(arguments)
+                .shr(&right.refine_parameters(arguments)),
+            ExpressionDomain::ShrOverflows {
+                left,
+                right,
+                result_type,
+            } => left
+                .refine_parameters(arguments)
+                .shr_overflows(&right.refine_parameters(arguments), result_type.clone()),
+            ExpressionDomain::Sub { left, right } => left
+                .refine_parameters(arguments)
+                .sub(&right.refine_parameters(arguments)),
+            ExpressionDomain::SubOverflows {
+                left,
+                right,
+                result_type,
+            } => left
+                .refine_parameters(arguments)
+                .sub_overflows(&right.refine_parameters(arguments), result_type.clone()),
+            ExpressionDomain::Variable { path, .. } => match **path {
+                Path::LocalVariable { ordinal } if 0 < ordinal && ordinal <= arguments.len() => {
```

Comment:
Did you mean use Yoda ordering on the '0 < ordinal'?

Reply:
Yes, I like to use 0 < x & x < n as an approximation for 0 < x < n.